File: chor-extraction/src/main/java/ast/sp/nodes/ConditionSP.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;
import ast.sp.nodes.expr.BooleanExpression;

public class ConditionSP implements Behaviour{
    private final String process;
    private final BooleanExpression expression;
    private final Behaviour thenBehaviour;
    private final Behaviour elseBehaviour;

    public ConditionSP(String process, BooleanExpression expression, Behaviour thenBehaviour, Behaviour elseBehaviour) {
        this.process = process;
        this.expression = expression;
        this.thenBehaviour = thenBehaviour;
        this.elseBehaviour = elseBehaviour;
    }

    public String getProcess() {
        return process;
    }

    public BooleanExpression getExpression() {
        return expression;
    }

    public Behaviour getThenBehaviour() {
        return thenBehaviour;
    }

    public Behaviour getElseBehaviour() {
        return elseBehaviour;
    }

    public String toString() {
        return "if " + process + "." + expression + " then " + thenBehaviour.toString() + " else " + elseBehaviour.toString();
    }

    @Override
    public boolean findRecProcCall(String procname) {
        return thenBehaviour.findRecProcCall(procname) || elseBehaviour.findRecProcCall(procname);
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/ConditionSP.kt
package ast.sp.nodes

import ast.sp.interfaces.Behaviour

data class ConditionSP(val expression: String, val thenBehaviour: Behaviour, val elseBehaviour: Behaviour) : Behaviour {
    override fun toString(): String {
        return "if " +  expression + " then " + thenBehaviour.toString() + " else " + elseBehaviour.toString()
    }

    override fun findRecProcCall(procname: String): Boolean {
        return thenBehaviour.findRecProcCall(procname) || elseBehaviour.findRecProcCall(procname)
    }

    override fun copy(): Behaviour {
        return ConditionSP(expression, thenBehaviour.copy(), elseBehaviour.copy())
    }

    override fun equals(b: Behaviour): Boolean {
        if (b is ConditionSP){
            return expression == b.expression &&
                    thenBehaviour.equals(b.thenBehaviour) &&
                    elseBehaviour.equals(b.elseBehaviour)
        }
        else return false
    }
}
